pkg/util/encryption/multi.go

```diff
@@ -28,7 +36,8 @@ func NewMulti(ctx context.Context, serviceKeyvault keyvault.Manager, secretName,
 	}
 
 	m := &multi{
-		sealer: aead,
+		sealer:      aead,
+		sealVersion: filepath.Base(*secret.ID),
```

Comment:
To be honest I am lost with how this PR should work. 
Srinivas would you please provide a detailed explanation of the logical flow and how this is supposed to work?